lib/model-utils.js

```diff
@@ -495,6 +462,45 @@ ModelUtils._coerce = function(where, options, modelDef) {
       }
     }
 
+    if (operator === 'regexp' && val instanceof RegExp) {
+      where[p] = val;
+      return where[p];
+    }
```

Comment:
The code handling `operator === 'regexp'` belongs `case 'regexp'` above.

IIUC, you are trying to update `where[p]` with the coerced value in case `DataType` is not defined. This is something we need to do for all operators (`regexp`, `inq`, etc.), not only for regular expressions. Please rework this block accordingly.

Personally, I think we should remove lines 465-468 above and update the block below as follows:

```diff
 let DataType = props[p] && props[p].type;
 if (!DataType) {
+  where[p] = val;
   continue;
 }
```

Reply:
Thanks @bajtos the reason I went with the above approach above is because I noticed accommodating other types was breaking other tests such as:

```mocha
1) basic-querying
       find
         query with nested property
           should support nested property with gt in query:

      Uncaught AssertionError: expected 0 to be 2
```